packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,632 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+
+/// Signature for a function that creates a [Widget] to be used within an
+/// [OpenContainer].
+///
+/// The `action` callback provided to [OpenContainer.closedBuilder] can be used
+/// to open the container. The `action` callback provided to
+/// [OpenContainer.closedBuilder] can be used to close the container again.
+typedef OpenContainerChildBuilder = Widget Function(
```

Comment:
Maybe just call this OpenContainerBuilder? That would forgo any discussion about if what was being built was actually the container's child, or a descendant.